internal/keygen/keygen.go

```diff
@@ -160,3 +174,77 @@ func generateEDDSAKey(tssService tss.Service, kg *types.KeyGeneration, partiesJo
 	time.Sleep(time.Second)
 	return resp, nil
 }
+
+func BackupVault(kg *types.KeyGeneration, partiesJoined []string, ecdsaPubkey, eddsaPubkey string, localStateAccessor *relay.LocalStateAccessorImp) error {
+	ecdsaKeyShare, err := localStateAccessor.GetLocalState(ecdsaPubkey)
+	if err != nil {
+		return fmt.Errorf("failed to get local sate: %w", err)
+	}
+
+	eddsaKeyShare, err := localStateAccessor.GetLocalState(eddsaPubkey)
+	if err != nil {
+		return fmt.Errorf("failed to get local sate: %w", err)
+	}
+
+	vault := &vaultType.Vault{
+		Name:           kg.Name,
+		PublicKeyEcdsa: ecdsaPubkey,
+		PublicKeyEddsa: eddsaPubkey,
+		Signers:        partiesJoined,
+		CreatedAt:      timestamppb.New(time.Now()),
+		HexChainCode:   kg.ChainCode,
+		KeyShares: []*vaultType.Vault_KeyShare{
+			{
+				PublicKey: ecdsaPubkey,
+				Keyshare:  ecdsaKeyShare,
+			},
+			{
+				PublicKey: eddsaPubkey,
+				Keyshare:  eddsaKeyShare,
+			},
+		},
+		LocalPartyId:  kg.Key,
+		ResharePrefix: "",
+	}
+
+	isEncrypted := kg.EncryptionPassword != ""
+	vaultData, err := proto.Marshal(vault)
+	if err != nil {
+		return fmt.Errorf("failed to Marshal vault: %w", err)
+	}
+
+	if isEncrypted {
+		vaultData, err = common.EncryptVault(kg.EncryptionPassword, vaultData)
+		if err != nil {
+			return fmt.Errorf("common.EncryptVault failed: %w", err)
+		}
+	}
+	vaultBackup := &vaultType.VaultContainer{
+		Version:     1,
+		Vault:       base64.StdEncoding.EncodeToString(vaultData),
+		IsEncrypted: isEncrypted,
+	}
+	filePathName := filepath.Join(config.AppConfig.Server.VaultsFilePath, kg.Name+".bak")
```

Comment:
can you please update it save the vault backup using `publickey_ecdsa.bak` ? so we can use the public key ecdsa to identify the backups?

It can be a seperate PR